src/services/services.ts

```diff
@@ -1405,8 +1405,10 @@ namespace ts {
 
         function dispose(): void {
             if (program) {
+                // Use paths to ensure we are using correct key and paths as document registry could bre created with different current directory than host
+                const key = documentRegistry.getKeyForCompilationSettings(program.getCompilerOptions());
                 forEach(program.getSourceFiles(), f =>
-                    documentRegistry.releaseDocument(f.fileName, program.getCompilerOptions()));
+                    documentRegistry.releaseDocumentWithKey(f.resolvedPath, key));

```

Comment:
Is this a new approach or just a bug fix?  Is the key already used when the document is added to the registry?

Reply:
This is not new approach as you see the method already exists and if you browse of other usages of methods of DocumentRegistry you would notice that they all use path and key.